FinSearchUnified/Tests/Bundle/ProductNumberSearchTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace FinSearchUnified\Tests\Bundle;
+
+use Enlight_Controller_Front as Front;
+use Enlight_Controller_Request_RequestHttp as RequestHttp;
+use FinSearchUnified\Bundle\ProductNumberSearch;
+use FinSearchUnified\Bundle\SearchBundleFindologic\QueryBuilder;
+use FinSearchUnified\Bundle\SearchBundleFindologic\QueryBuilderFactory;
+use Shopware\Bundle\SearchBundle\Criteria;
+use Shopware\Components\Test\Plugin\TestCase;
+use SimpleXMLElement;
+
+class ProductNumberSearchTest extends TestCase
+{
+    protected function tearDown()
+    {
+        parent::tearDown();
+        Shopware()->Container()->reset('front');
+        Shopware()->Container()->load('front');
+        Shopware()->Session()->offsetUnset('findologicDI');
+        Shopware()->Session()->offsetUnset('isSearchPage');
+    }
+
+    /**
+     * @dataProvider productNumberSearchProvider
+     *
+     * @param bool $isFetchCount
+     * @param bool $isUseShopSearch
+     * @param string|null $response
+     * @param int $invokationCount
+     *
+     * @throws \Exception
+     */
+    public function testProductNumberSearchImplementation($isFetchCount, $isUseShopSearch, $response, $invokationCount)
+    {
+        $criteria = new Criteria();
+        $criteria->setFetchCount($isFetchCount);
+
+        Shopware()->Session()->findologicDI = $isUseShopSearch;
+        Shopware()->Session()->isSearchPage = !$isUseShopSearch;
+
+        $mockedQuery = $this->getMockBuilder(QueryBuilder::class)
+            ->disableOriginalConstructor()
+            ->setMethods(['execute'])
+            ->getMockForAbstractClass();
+
+        if ($response === 'xml') {
+            $data = '<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>';
```

Comment:
Generating the XML could also be done in the provider itself, so you don't have to perform this check.